powerscale/provider/storagepool_tier_resource.go

```diff
@@ -0,0 +1,263 @@
+func (r *StoragepoolTierResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	tflog.Info(ctx, "updating storagepool settings resource state")
+	var plan *models.StoragepoolTierResourceModel
+	var state models.StoragepoolTierResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read the state
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// Populate the Edit Parameters
+	editValues := powerscale.V16StoragepoolTierExtendedExtended{}
+	if state.Name != plan.Name {
+		editValues.Name = plan.Name.ValueStringPointer()
+	}
+
+	if !state.Children.Equal(plan.Children) {
+		var ChildrenList []string
+		if diags := plan.Children.ElementsAs(ctx, &ChildrenList, false); diags.HasError() {
+			return
+		}
+		editValues.Children = append(make([]string, 0), ChildrenList...)
+	}
+
+	updatedThroughCondition := false
+	if !plan.TransferLimitState.IsNull() && plan.TransferLimitState.ValueString() != "" && state.TransferLimitState != plan.TransferLimitState {
+		editValues.TransferLimitState = plan.TransferLimitState.ValueStringPointer()
+		if plan.TransferLimitPct.IsNull() || plan.TransferLimitPct.ValueInt64() == 0 {
+			editValues.TransferLimitPct = nil
+			updatedThroughCondition = true
+		}
+	}
+
+	if !updatedThroughCondition && state.TransferLimitPct != plan.TransferLimitPct {
+		value := float64(plan.TransferLimitPct.ValueInt64())
+		editValues.TransferLimitPct = &value
+		if plan.TransferLimitState.IsNull() || plan.TransferLimitState.IsUnknown() {
+			editValues.TransferLimitState = nil
+		}
+	}
+
+	diags := helper.UpdateStoragepoolTier(ctx, r.client, editValues, &state)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	diags = helper.ReadStoragepoolTier(ctx, r.client, &state)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	tflog.Info(ctx, "Done with Update Storagepool Tier resource state")
+}
+
+// Delete deletes the resource.
+func (r *StoragepoolTierResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	tflog.Info(ctx, "deleting Storagepool tier")
+	var data *models.StoragepoolTierResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	deleteParam := r.client.PscaleOpenAPIClient.StoragepoolApi.DeleteStoragepoolv16StoragepoolTier(ctx, strconv.FormatInt(data.Id.ValueInt64(), 10))
+	_, err := deleteParam.Execute()
+	if err != nil {
+		errStr := constants.DeleteStoragepoolTierErrorMsg + "with error: "
+		message := helper.GetErrorString(err, errStr)
+		resp.Diagnostics.AddError(
+			"Error deleting storage pool",
+			message,
+		)
+	}
```

Comment:
Is state getting cleared after deletion?

Reply:
yes